### PR TITLE
Don't initialize trace if specifically asked not to

### PR DESCRIPTION
**Pull request by mayankbansal018**

### CODE REVIEW

**Comment by mayankbansal018** [comment]
@dotnet-bot test this please

**Comment by mayankbansal018** [comment]
@singhsarab why did this got closed?

File: src/testhost.x86/DefaultEngineInvoker.cs [hunk 1]
```diff
@@ -49,12 +49,14 @@ internal class DefaultEngineInvoker :
...
-            string logFile;
-            if (argsDictionary.TryGetValue(LogFileArgument, out logFile))
+            if (argsDictionary.TryGetValue(LogFileArgument, out string logFile))
             {
                 EqtTrace.InitializeVerboseTrace(logFile);
             }
-
+            else
+            {
+                EqtTrace.DoNotInitailize = true;
```
Comment (abhishkk) [inline src/testhost.x86/DefaultEngineInvoker.cs]: Should we not do the same for data collectors and vstest console processes?

Comment (abhishkk) [inline src/testhost.x86/DefaultEngineInvoker.cs]: Unit tests for the case when DoNotInitialize is true to ensure trace is not initialzied?

**Review by abhishkk — APPROVED** [review]

**Review by smadala — APPROVED** [review]

File: src/Microsoft.TestPlatform.CoreUtilities/Tracing/EqtTrace.cs [hunk 2]
```diff
@@ -72,6 +72,18 @@ public static string LogFile
             }
         }
 
+        public static bool DoNotInitailize
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.CoreUtilities/Tracing/EqtTrace.cs]: Nit: DoNotInitialize

**Pull request merged**